### PR TITLE
Add password text field

### PR DESCRIPTION
**Pull request by kelianClerc**
EDIT: after the following video I've inverted the eye icon state. non barred one for password visible
https://user-images.githubusercontent.com/26220096/221232906-dfb088d3-502f-4ec2-9402-24c1183504ab.mov

### CODE REVIEW

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt [hunk 1]
```diff
@@ -35,6 +39,74 @@ import com.spendesk.grapes.compose.theme.GrapesTheme
...
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    singleLine: Boolean = false,
+    colors: GrapesTextFieldColors = GrapesTextFieldDefaults.textFieldColors(),
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    leadingIcon: @Composable (() -> Unit)? = null,
+    trailingIcon: @Composable (() -> Unit)? = null,
+) {
+    // Holds the latest internal TextFieldValue state. We need to keep it to have the correct value
+    // of the composition.
```
Comment (Tweener) [inline library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt:63]: Check your line characters limit, it looks like it's less than 200? (Lines are split too soon)

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt [hunk 2]
```diff
@@ -77,6 +77,43 @@ fun GrapesTextInput(
     )
 }
 
+@ExperimentalMaterial3Api
+@Composable
+fun GrapesTextInput(
```
Comment (Tweener) [inline library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt:82]: Why this public Composable has `Grapes` as a prefix, whereas `PasswordTextInput` doesn't?
We should have a guildeline about the naming convention for "public composables".

**Review by Tweener — COMMENTED** [review]
I commented mostly about guidelines we might not have yet and need to define altogether (naming convention, orders of Composable parameters, etc.).
Depending on what we decide, I'll approve this PR!

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt [hunk 3]
```diff
@@ -35,6 +39,74 @@ import com.spendesk.grapes.compose.theme.GrapesTheme
  * @since 06/01/2023, Fri
  **/
 
+@ExperimentalMaterial3Api
+@Composable
+internal fun GrapesBaseTextField(
+    value: String,
+    placeholderValue: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
```
[the remark this answers is not in the document]
Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt:48]: Nop it should be the first of optional parameters. You don't necessarily provide a modifier, whereas you are forced to provide non optional parameters.
If not naming parameters, you would have: 
`Comp(a, b)`
which is better than
`Comp(Modifier, a, b)` (empty default modifier)

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt (hunk 1 above)
Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt:63]: It's a copy paste of material code. I may add a comment saying so before all that. 
Is it ok for you to keep it like that to compare easily the copy paste code?

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt (hunk 2 above)
Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt:82]: In my mind it was: 
Basic composable are prefixed Grapes, and composables built on top of basic ones, not necessarly.
But we could add a point in the chapter to settle something

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tweener) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt:48]: Alright, fair enough! Thanks for the explanation :)

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt (hunk 1 above)
Comment (Tweener) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesBaseTextField.kt:63]: Sure, no worries

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt (hunk 2 above)
Comment (Tweener) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesTextInput.kt:82]: Yep agreed to add it to the next chapter where we are all here.

**Review by Tweener — APPROVED** [review]

**Pull request merged**